ui/apps/platform/src/Containers/Clusters/ClustersTable.tsx

```diff
@@ -0,0 +1,246 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { Flex } from '@patternfly/react-core';
+import {
+    ActionsColumn,
+    ExpandableRowContent,
+    Table,
+    Tbody,
+    Td,
+    Th,
+    Thead,
+    Tr,
+} from '@patternfly/react-table';
+
+import TbodyUnified from 'Components/TableStateTemplates/TbodyUnified';
+import useMetadata from 'hooks/useMetadata';
+import { Cluster } from 'types/cluster.proto';
+import { ClusterIdToRetentionInfo } from 'types/clusterService.proto';
+import { TableUIState } from 'utils/getTableUIState';
+
+import { formatCloudProvider } from './cluster.helpers';
+import { CertExpiryStatus, ClusterHealthStatus } from './clusterTypes';
+import ClusterDeletion from './Components/ClusterDeletion';
+import ClusterStatus from './Components/ClusterStatus';
+import CredentialExpiration from './Components/CredentialExpiration';
+import HelmIndicator from './Components/HelmIndicator';
+import OperatorIndicator from './Components/OperatorIndicator';
+import SensorUpgrade from './Components/SensorUpgrade';
+
+export type ClustersTableProps = {
+    clusterIdToRetentionInfo: ClusterIdToRetentionInfo;
+    tableState: TableUIState<Cluster>;
+    selectedClusterIds: string[];
+    onClearFilters: () => void;
+    onDeleteCluster: (cluster: Cluster) => (event: React.MouseEvent) => void;
+    toggleAllClusters: () => void;
+    toggleCluster: (clusterId) => void;
+};
+
+export const COL = {
+    STATUS: 'status',
+    SENSOR: 'sensor',
+} as const;
+type ColId = (typeof COL)[keyof typeof COL];
+
+type ExpansionMap = Record<string, ColId | null>;
+
+function ClustersTable({
+    clusterIdToRetentionInfo,
+    tableState,
+    selectedClusterIds,
+    onClearFilters,
+    onDeleteCluster,
+    toggleAllClusters,
+    toggleCluster,
+}: ClustersTableProps) {
+    const metadata = useMetadata();
```

Comment:
nit: Completely up to you, but we could move `const metadata = useMetadata()` up to the parent and pass it down as a prop. That’d make this component a bit more independent from the Redux store and easier to reuse or test if needed.

Reply:
Yeah, `ClustersTablePanel` already has it. And `version: string` is only property used

Reply:
Good idea, went ahead and pull that hook out and added the `version` prop.